### PR TITLE
Add Closure support

### PR DESCRIPTION
**Pull request by jiayliu**
How to use:
Run setup-closure.sh from the same dir as apprtc.py. This will download the closure library and compiler.
Run closure-build.sh to compile the files in closure-src. The output is in closure-out as compile.js, which can be referenced from the HTML file in a script tag.

closure-src/ only has a utils.js for now. We can convert the other pieces gradually.

@phoglund

### CODE REVIEW

**Comment by KaptenJansson** [comment]
ping phoglund

File: samples/web/content/apprtc/setup-closure.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+rm -rf closure-library
+echo "Downloading Closure library:"
+git clone https://github.com/google/closure-library
+echo "Downloading Closure library done."
+
+echo "Downloading Closure compiler:"
+curl -o compiler.zip http://dl.google.com/closure-compiler/compiler-latest.zip
```
Comment (phoglund) [inline samples/web/content/apprtc/setup-closure.sh]: Consider passing -s or -sS here depending on how verbose the curl output is.

**Comment by phoglund** [comment]
First, sorry, I completely forgot about this one. Are we still going ahead with this?

I think this should work on the Travis boxes since everything is downloaded on the fly. You can try it by hooking up a shell job which invokes your .sh in the gruntfile (like for the python tests). Then make a new commit and push to your branch. Travis will run on that (similarly to the green checkbox next to your Add Closure Support commit above) and you will see if it works.

How will this work then, will one have to run the .sh before bringing up the apprtc server so all the closure stuff gets generated?

**Comment by juberti** [comment]
I think it makes sense to do this so that we can download a single minified JS lib, but ideally it would be possible to run unoptimized (i.e. what we have now) as well.

**Comment by jiayliu** [comment]
We added a 'build' task to grunt for packaging the Chrome App. We can add
the closure build script to the build task.

We can also use closure to generate unoptimized code into one JS file for
debugging.

I'll update this patch to make that happen.

On Tue, Jan 20, 2015 at 12:00 PM, Justin Uberti <notifications@github.com>
wrote:

> I think it makes sense to do this so that we can download a single
> minified JS lib, but ideally it would be possible to run unoptimized (i.e.
> what we have now) as well.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/webrtc/pull/303#issuecomment-70721949>.
>



-- 
Jiayang Liu

**Comment by jiayliu** [comment]
I'll create a new PR since this one is compared against an old branch.

**Pull request closed without merging**